pkg/resource/clickpipe.go

```diff
@@ -796,6 +810,44 @@ func (c *ClickPipeResource) ModifyPlan(ctx context.Context, request resource.Mod
 		}
 	}
 
+	// Validate queue_url configuration for object storage
+	if !plan.Source.IsNull() {
+		sourceModel := models.ClickPipeSourceModel{}
+		response.Diagnostics.Append(plan.Source.As(ctx, &sourceModel, basetypes.ObjectAsOptions{})...)
+
+		if !sourceModel.ObjectStorage.IsNull() {
+			objectStorageModel := models.ClickPipeObjectStorageSourceModel{}
+			response.Diagnostics.Append(sourceModel.ObjectStorage.As(ctx, &objectStorageModel, basetypes.ObjectAsOptions{})...)
+
+			// Validate queue_url is only provided when is_continuous is true
+			if !objectStorageModel.QueueURL.IsNull() && !objectStorageModel.QueueURL.IsUnknown() && objectStorageModel.QueueURL.ValueString() != "" {
```

Comment:
you can and imho should do these checks in the schema definition above using AlsoRequires, ConflictsWith and the like. See https://developer.hashicorp.com/terraform/plugin/framework/migrating/attributes-blocks/validators-predefined

Reply:
I trust your Terraform expertise far more than my own, but these validations check specific field values (e.g. `type == "s3"`) across nested attributes, not just presence. Predefined validators like `AlsoRequires` seems to work best for simple "field A needs field B" cases (correct if I'm wrong, please). IMO, this business logic is clearer and more maintainable in `ModifyPlan` with explicit error messages.

Does that make sense to you? :)